### PR TITLE
Provide support for JSON and JSONB datatypes.

### PR DESCRIPTION
**Pull request by HieuNguyen202**
Jira:
https://jsw.ibm.com/browse/NEXTGEN-1422

Problem:

Two new datatypes were/will be introduced to IPS: JSON (introduced in IPS 11.0.3.0) and JSONB (to be introduced in soon). The support of these two datatypes on the server-side requires some changes on the client-side so it can retrieve JSON/JSONB data from the server. Specifically, there are two major changes required:

Solution:

1. Enabling nzgo client to send its client version string `Release 11.1.0.0`' to the server at the beginning of each session, which tells the server that nzgo client is capable of handling JSON/ JSONB data. Notice: the client version string must start with a string, such as 'Release', in order for the server to properly parse it.
2. Mapping JSON/JSONB data to string type in nzgo, which can be handled in a similar fashion as NVARCHAR data in nzgo.

Testing:

Tests can be invoked by `go test -run TestJson`, which triggers tests for both JSON and JSONB datatype.

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -136,35 +136,36 @@ const (
...
-	NzTypeInt8
-	NzTypeVarFixedChar
-	NzTypeGeometry
-	NzTypeVarBinary
-	NzDEPR_Blob // OBSOLETE 3.0: BLAST Era Large 'binary' Object
-	NzTypeNChar
-	NzTypeNVarChar
-	NzDEPR_NText    // OBSOLETE 3.0: BLAST Era Large 'nchar text' Object
-	NzTypeLastEntry // KEEP THIS ENTRY LAST - used internally to size an array
+	NzTypeRecAddr      = 1 // !NOTE-bmz need to add this to all switch stmts
```
Comment (rafoid) [inline conn.go:139]: why explicitly define every constant instead of using **iota**?

File: testInterfaces_test.go [hunk 2]
```diff
@@ -24,10 +24,14 @@ const (
 )
 
 func openTestConnConninfo(conninfostr string) (*sql.DB, error) {
+	elog := PDALogger{"DEBUG", ""}
```
Comment (rafoid) [inline testInterfaces_test.go:27]: do we need elog init in every function?

File: conn.go (hunk 1 above)
Comment (HieuNguyen202) [reply conn.go:139]: Because auto-increment of the number would result in `NzTypeJson = 28` and `NzTypeJson = 29`, which are not the same as what we have in the backend ( `NzTypeJson = 30` and `NzTypeJson = 31`).

File: testInterfaces_test.go (hunk 2 above)
Comment (HieuNguyen202) [reply testInterfaces_test.go:27]: Yes, we do. There was a recent change that requires all applications using nzgo driver to init elog, else they will get a segmentation fault when trying to connect to the backend. In this test file, every test function can be independently invoked with a `go test` command, so I had to intit elog everywhere.

File: conn.go (hunk 1 above)
Comment (HieuNguyen202) [reply conn.go]: It's been changed back to auto enumeration with the blank identifier `_` being used as a placeholder for the missing enums.

File: version.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
 package nzgo
 
-// this is version of nzgo driver
-const nzgo_client_version = "11.0.0.0"
+// this is the version of nzgo driver
```
Comment (abhishekjog) [inline version.go:3]: please revert this change

Comment (HieuNguyen202) [reply version.go]: It's been reverted
